spv/sync.go

```diff
@@ -800,6 +916,9 @@ func (s *Syncer) handleTxInvs(ctx context.Context, rp *p2p.RemotePeer, hashes []
 			op := errors.Opf(opf, rp.RemoteAddr())
 			log.Warn(errors.E(op, err))
 		}
+		if s.checkTSpend(ctx, tx) {
+			s.wallet.AddTSpend(*tx)
+		}
```

Comment:
Was doing another functional test for this case and I realized this section is only called if the tspend has an output relevant to the wallet.

This needs to be moved into a loop of its own, before filtering for relevant transactions, otherwise the wallet will not know about the tspend to set a voting policy for.